src/main/java/org/dependencytrack/tasks/OSVDownloadTask.java

```diff
@@ -0,0 +1,202 @@
+import alpine.model.ConfigProperty;
+import kong.unirest.json.JSONObject;
+import org.apache.http.StatusLine;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.methods.HttpUriRequest;
+import org.dependencytrack.common.HttpClientPool;
+import org.dependencytrack.event.GoogleOSVMirrorEvent;
+import org.dependencytrack.event.IndexEvent;
+import org.dependencytrack.model.Cwe;
+import org.dependencytrack.model.Severity;
+import org.dependencytrack.model.Vulnerability;
+import org.dependencytrack.model.VulnerableSoftware;
+import org.dependencytrack.parser.common.resolver.CweResolver;
+import org.dependencytrack.parser.osv.GoogleOSVAdvisoryParser;
+import org.dependencytrack.parser.osv.model.Ecosystem;
+import org.dependencytrack.parser.osv.model.OSVAdvisory;
+import org.dependencytrack.parser.osv.model.OSVVulnerability;
+import org.dependencytrack.persistence.QueryManager;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipInputStream;
+
+import static org.dependencytrack.model.ConfigPropertyConstants.VULNERABILITY_SOURCE_GOOGLE_OSV_ENABLED;
+
+public class OSVDownloadTask implements LoggableSubscriber {
+
+    private static final Logger LOGGER = Logger.getLogger(OSVDownloadTask.class);
+    private final boolean isEnabled;
+    private HttpUriRequest request;
+
+    public OSVDownloadTask() {
+        try (final QueryManager qm = new QueryManager()) {
+            final ConfigProperty enabled = qm.getConfigProperty(VULNERABILITY_SOURCE_GOOGLE_OSV_ENABLED.getGroupName(), VULNERABILITY_SOURCE_GOOGLE_OSV_ENABLED.getPropertyName());
+            this.isEnabled = enabled != null && Boolean.valueOf(enabled.getPropertyValue());
+        }
+    }
+
+    @Override
+    public void inform(Event e) {
+
+        if (e instanceof GoogleOSVMirrorEvent && this.isEnabled) {
+
+            try {
+                for (Ecosystem ecosystem : Ecosystem.values()) {
+                    LOGGER.info("Updating datasource with Google OSV advisories for ecosystem " + ecosystem.getValue());
+                    String url = "https://osv-vulnerabilities.storage.googleapis.com/" + ecosystem.getValue() + "/all.zip";
+                    request = new HttpGet(url);
+                    try (final CloseableHttpResponse response = HttpClientPool.getClient().execute(request)) {
+                        final StatusLine status = response.getStatusLine();
+                        if (status.getStatusCode() == 200) {
+                            try (InputStream in = response.getEntity().getContent()) {
+                                ZipInputStream zipInput = new ZipInputStream(in);
+                                unzipFolder(zipInput);
+                            }
```

Comment:
`zipInput` should be closed as well.

```suggestion
                            try (InputStream in = response.getEntity().getContent();
                                 ZipInputStream zipInput = new ZipInputStream(in)) {
                                unzipFolder(zipInput);
                            }
```